src/app/search/loadouts/search-filters/freeform.ts

```diff
@@ -91,36 +84,38 @@ const freeformFilters: FilterDefinition<
     },
   },
   {
-    keywords: 'contains',
+    keywords: ['contains', 'exactcontains'],
     description: tl('LoadoutFilter.Contains'),
     format: 'freeform',
-    suggestionsGenerator: ({ d2Definitions, loadouts, selectedLoadoutsStore }) => {
+    suggestionsGenerator: ({ d2Definitions, loadouts }) => {
       if (!d2Definitions || !loadouts) {
         return [];
       }
 
       return loadouts.flatMap((loadout) => {
-        if (loadout.classType !== selectedLoadoutsStore?.classType) {
-          return [];
-        }
-
         const itemSuggestions = loadout.items.map((item) => {
           const definition = d2Definitions.InventoryItem.get(item.hash);
-          return `contains:${quoteFilterString(definition.displayProperties.name.toLowerCase())}`;
+          return `exactcontains:${quoteFilterString(definition.displayProperties.name.toLowerCase())}`;
         });
         const modSuggestions =
           loadout.parameters?.mods?.map((modHash) => {
             const definition = d2Definitions.InventoryItem.get(modHash);
-            return `contains:${quoteFilterString(definition.displayProperties.name.toLowerCase())}`;
+            // I am not sure why, but this definition can sometimes be undefined even though the modHash
+            // looks accurate. It only seems to happen after lifting the loadout filtering on classType
+            // up to the context creation scope.
+            return (
+              definition &&
+              `exactcontains:${quoteFilterString(definition.displayProperties.name.toLowerCase())}`
+            );
```

Comment:
Another weird thing that is happening as a result of these changes.

Reply:
Actually this broke beta, reverting the PR that introduced it so I can raise it again

Reply:
You figured it out, but it's because of mod resolution.